### ISSUE TITLE
False positive when an `Optional` is unpacked with `None not in`.

### ISSUE THREAD

**Issue by yanqd0** [python/mypy#11339]
**Bug Report**

False positive when an `Optional` is unpacked with `None not in`.

**To Reproduce**

`demo.py`:

```python
from typing import Optional

a: Optional[int] = 1
b: Optional[int] = 1

if None not in (a, b):
    print(a + b)
```

**Actual Behavior**

```shell
$ mypy demo.py 
demo.py:7: error: Unsupported operand types for + ("int" and "None")
demo.py:7: error: Unsupported operand types for + ("None" and "int")
demo.py:7: error: Unsupported left operand type for + ("None")
demo.py:7: note: Both left and right operands are unions
Found 3 errors in 1 file (checked 1 source file)
```

**Expected Behavior**

```shell
$ mypy demo.py 
Success: no issues found in 1 source file
```

In fact, the 2 expression should be the same:

```python
# mypy failed
if None not in (a, b):
    print(a + b)

# mypy passed
if a is not None and b is not None:
    print(a + b)
```

**Your Environment**

```shell
$ mypy --version
mypy 0.910
$ python --version
Python 3.8.10
$ uname -a
Linux yan-p520c 5.4.0-88-generic #99-Ubuntu SMP Thu Sep 23 17:29:00 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.3 LTS
Release:        20.04
Codename:       focal
```

**Comment by AlexWaygood** [comment]
Duplicate of #2980
